src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java

```diff
@@ -0,0 +1,291 @@
+package hex.genmodel.easy;
```

Comment:
Sorry for the long time this answer took me :/ It turned out to be a convoluted problem and I had to do some tests and investigation.
Model version independence is a good idea and proper approach, but unfortunately it's not possible to obtain (at least without some really strange solutions) due to incompatible changes in h2o-genmodel library.
All models generated by H2O server are extensions of hex.genmodel.GenModel class from h2o-genmodel. Between 3.0.x and 3.1.0 there was an incompatible change in this class. All generated model POJOs implement score0(double[], double[]) method and some of them, in a body of this method, call getPrediction(...) method, which is already implemented in GenModel parent class. Before 3.1.x there was getPrediction(double[], double, double) method, but in 3.1.0 it was removed and getPrediction(double[], double[], double, double) was introduced. So if you have h2o-genmodel.jar in a version prior to 3.1.0 than all newer models will not work with it, and opposite - h2o-genmodel in 3.1.0 and higher will break older models, generated with 3.0.x versions. Looks like other people also noticed it and are not happy: https://github.com/h2oai/h2o-3/commit/f8b188e4775d0f3671c34b3c42fe9c417d960cfd.
In order to keep model version independence the code in scoring-engine-base would have to recognize model version and adjust it's runtime to it by making some changes in classloader, in order to provide proper version of h2o-genmodel library. And model version detection is not easy - h2o-genmodel library downloaded from h2o-server has no information about it's version. So we would have to parse POJO class and make decision based on getPrediction method signature... Sounds really bad.

Another thing is that your code uses classes from hex.genmodel.easy package which was added also in 3.1.0 version of h2o-genmodel. This is another reason to not keep backwards compatibility - a code that uses those classes will not work with older models.

So taking into consideration the two things above as well as the fact, that now we can live without backward compatibility, I've decided to break it.

I've just tagged last master commit with "3.0.x_models_compatible" and from this place all commits will support only 3.1.x and higher model versions, so now we can go ahead with your Pull Request.
So the next step is to update h2o-genmodel version in pom.xml and replace existing test model with the one compatible with 3.1.x and higher versions of library. I've just pushed a branch "test_model_compatible_with_3_1_x" with new test model and updated pom.xml - it is the same model as current one, just generated on a newer H2O server. You can take new model from it. In case you would rather generate new model by yourself, here's how to do it:

1. create some model on your h2o server (model currently attached in test resources is a simple GLM model based on Iris flower data set taken from here: http://archive.ics.uci.edu/ml/machine-learning-databases/iris/, response column is set to class column)
2. download model from server: ```
wget http://<h2o-server-url>/3/Models.java/<model name>```
3. download h2o-genmodel library from server```
wget http://<h2o-server-url>/3/h2o-genmodel.jar```
4. rename model file to <model_name>.java
5. compile the model with the library```
javac -cp h2o-genmodel.jar <model_name>.java -d <dir_for_classes>```
6. build jar file from compiled .class files```
jar cvf h2o-model.jar -C <dir_for_classes> .```
7. after attaching new model to test resources, adjust given and expected numbers in integration test to the new model if necessary

Sorry again for the long time you had to wait.